lua/lspconfig/server_configurations/rust_analyzer.lua

```diff
@@ -10,6 +10,40 @@ local function reload_workspace(bufnr)
   end)
 end
 
+local function get_workspace_dir(cmd)
+  local co = assert(coroutine.running())
+
+  local chunks = {}
+  local jobid = vim.fn.jobstart(cmd, {
+    on_stdout = function(_, data, _)
+      chunks[#chunks + 1] = unpack(data)
+    end,
+    on_stderr = function(_, data, _)
+      vim.api.nvim_err_write(table.concat(data, '\n'))
+    end,
+    on_exit = function()
+      if next(chunks) == nil then
+        coroutine.resume(co, nil)
+        return
+      end
+
+      local data = table.concat(chunks, '')
+      chunks = vim.json.decode(data)
+      local workspace_root = chunks and chunks['workspace_root'] or nil
```

Comment:
Can you move all of this out of the callback function, after `coroutine.yield()` and just call `coroutine.resume(co)` here?